src/test/clojure/clara/test_durability.clj

```diff
@@ -254,4 +306,72 @@
             (check-fact expected-fact fact)))))))
  
 (deftest test-durability-fressian-serde
-  (durability-test :fressian))
+  (testing "SerDe of the rulebase along with working memory"
+    (durability-test :fressian))
+
+  (testing "Repeated SerDe of rulebase"
+    (let [rb-serde (fn [s]
+                     (with-open [baos (java.io.ByteArrayOutputStream.)]
+                       (d/serialize-rulebase s (df/create-session-serializer baos))
+                       (let [rb-data (.toByteArray baos)]
+                         (with-open [bais (java.io.ByteArrayInputStream. rb-data)]
+                           (d/deserialize-rulebase (df/create-session-serializer bais))))))
+
+          s (mk-session 'clara.durability-rules)
+          rb (-> s eng/components :rulebase)
+          deserialized1 (rb-serde s)
+          ;; Need a session to do the 2nd round of SerDe.
+          restored1 (d/assemble-restored-session deserialized1 {})
+          deserialized2 (rb-serde restored1)
+          restored2 (d/assemble-restored-session deserialized2 {})
+
+          init-qresults (:query-results (session-test s))
+          restored-qresults1 (:query-results (session-test restored1))
+          restored-qresults2 (:query-results (session-test restored2))]
+      
```

Comment:
Yes.  Obviously it doesn't test total correctness of the session or anything, but gives reasonable confidence since the results end up coming out the same after inserting the test facts, firing, and querying.

The working memory style test uses this same test, but takes it a step further and checks the object-identity found within the query results etc.